### DISCUSSION TITLE
开机自动启动

### DISCUSSION THREAD

**Discussion by Muska-Ami** [Muska-Ami/NyaLCF#97] [Show and tell]
允许开机自动启动并设置自启动隧道

**Comment by ltzXiaoYanMo** [comment]
目前大概可行的办法：<br>
在`pubspec.yaml`添加代码（依赖项）
```yaml
dependencies:
  flutter:
    sdk: flutter
  flutter_windowmanager: ^0.1.4
```
在代码中可以这样实现：
```dart
void main() {
  runApp(NyaLCF());

  // 注册开机自启动
  registerStartup();
}

void registerStartup() async {
  try {
    await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_SECURE);
    await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_KEEP_SCREEN_ON);
    await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_SHOW_WALLPAPER);
    await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_TURN_SCREEN_ON);
    await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON);
  } catch (e) {
    print("Error registering startup: $e");
  }
}
```

**Comment by Muska-Ami** [comment]
> 目前大概可行的办法： 在`pubspec.yaml`添加代码（依赖项）
> 
> ```yaml
> dependencies:
>   flutter:
>     sdk: flutter
>   flutter_windowmanager: ^0.1.4
> ```
> 
> 在代码中可以这样实现：
> 
> ```dart
> void main() {
>   runApp(NyaLCF());
> 
>   // 注册开机自启动
>   registerStartup();
> }
> 
> void registerStartup() async {
>   try {
>     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_SECURE);
>     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_KEEP_SCREEN_ON);
>     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_SHOW_WALLPAPER);
>     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_TURN_SCREEN_ON);
>     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON);
>   } catch (e) {
>     print("Error registering startup: $e");
>   }
> }
> ```

![image](https://github.com/Muska-Ami/NyaLCF/assets/79651409/33184248-815b-4ecd-8586-87aace20889e)

**Comment by ltzXiaoYanMo** [comment]
> > 目前大概可行的办法： 在`pubspec.yaml`添加代码（依赖项）
> > 
> > ```yaml
> > dependencies:
> >   flutter:
> >     sdk: flutter
> >   flutter_windowmanager: ^0.1.4
> > ```
> > 
> > 在代码中可以这样实现：
> > 
> > ```dart
> > void main() {
> >   runApp(NyaLCF());
> > 
> >   // 注册开机自启动
> >   registerStartup();
> > }
> > 
> > void registerStartup() async {
> >   try {
> >     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_SECURE);
> >     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_KEEP_SCREEN_ON);
> >     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_SHOW_WALLPAPER);
> >     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_TURN_SCREEN_ON);
> >     await FlutterWindowManager.addFlags(FlutterWindowManager.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON);
> >   } catch (e) {
> >     print("Error registering startup: $e");
> >   }
> > }
> > ```
> 
> ![image](https://github.com/Muska-Ami/NyaLCF/assets/79651409/33184248-815b-4ecd-8586-87aace20889e)
> 

草你妈我上一个月发的玩意你现在才看到是吧（

**Comment by Muska-Ami** [comment]
@ltzXiaoYanMo 你不看我代码写的啥（

**Comment by ltzXiaoYanMo** [comment]
> @ltzXiaoYanMo 你不看我代码写的啥（

![image](https://github.com/Muska-Ami/NyaLCF/assets/115706851/f7fbc552-b0d4-4b3b-8115-61f70db79508)

**Comment by ltzXiaoYanMo** [comment]
我没意见（？

**Comment by Muska-Ami** [comment]
> 我没意见（？

https://github.com/Muska-Ami/NyaLCF/blob/main/lib/utils/frpc/startup_loader.dart
Byd是这个

**Comment by Muska-Ami** [comment]
![b14ff31aa0db8499b5c5abdb9bc0c47b](https://github.com/Muska-Ami/NyaLCF/assets/79651409/61b60207-a17d-490f-b59d-1955a7f7f62f)

**Comment by ltzXiaoYanMo** [comment]
![5fe4a80b7aebf6c8e03f2e990afc320c](https://github.com/Muska-Ami/NyaLCF/assets/115706851/1658ac6f-21a5-40c2-ae9b-9ac3d35eec75)

**Comment by Muska-Ami** [comment]
![ac312761d6887ecb3dc0a17e4cb78453](https://github.com/Muska-Ami/NyaLCF/assets/79651409/5629fa49-8223-4ff3-a2f5-e0293d860161)

**Comment by ltzXiaoYanMo** [comment]
![c88cdd3f8ce83a58ebf49661db4999fe](https://github.com/Muska-Ami/NyaLCF/assets/115706851/62ba17f3-8ebd-43ba-a6f0-b724a49203f2)
![c988a580f52a15c65b9c1aabde155c9e](https://github.com/Muska-Ami/NyaLCF/assets/115706851/a0be22fb-0539-4fd1-bb13-4504f4661231)
![53fad02e6e035a84ff3145fc2e90edad](https://github.com/Muska-Ami/NyaLCF/assets/115706851/eac6e0e9-2653-4a75-a717-15774eef91aa)

**Comment by ltzXiaoYanMo** [comment]
![1d77d73d3b888d7673803c0359f8dcbe](https://github.com/Muska-Ami/NyaLCF/assets/115706851/3b6c75c9-d7c8-44bd-a5ba-a1bd342bdbe0)
